File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 == 1:
            return False
        else:
            st = []
            symbol = s
            for index in range(len(s)): # s = "(){}[]"
                if s[index] == "(":
                    st.append(")")
                elif s[index] == "{":
                    st.append("}")
                elif s[index] == "[":
                    st.append("]")
                elif len(st) == 0 or st.pop() != s[index]:
                    return False
            return len(st) == 0
            
            
        

File: 20-valid-parentheses/20-valid-parentheses.java
import java.util.Stack;
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        int N = s.length();
        if (N % 2 != 0) {
            return false;
        }
        
        for (int i = 0; i < N; i++) {
            if (s.charAt(i) == '(') {
                st.push(')');
            }
            else if (s.charAt(i) == '{') {
                st.push('}');
            }
            else if(s.charAt(i) == '[') {
                st.push(']');
            }
            else if (st.isEmpty() || st.pop() != s.charAt(i)) {
                return false;                
            }
        }
        return st.isEmpty();
    }
}